### PR TITLE
[in-process] Fix error handling.

### PR DESCRIPTION
**Pull request by kzhuravl**
This patch fixes an internal OpenCL test.

### CODE REVIEW

**Comment by kzhuravl** [comment]
*During in-process linking.

**Comment by rampitec** [comment]
Does it work if FE in-process is turned back on (for example with AMD_OCL_IN_PROCESS)? It may conflict with the diagnostic printer already created for clang.

**Comment by kzhuravl** [comment]
> Does it work if FE in-process is turned back on (for example with AMD_OCL_IN_PROCESS)?
Yes.

> It may conflict with the diagnostic printer already created for clang.
I do not see how. The diagnostic handler is attached to the context that is created by the driver specifically for linking.

File: src/driver/AmdCompiler.cpp [hunk 1]
```diff
@@ -86,6 +88,39 @@ Verbose("v", cl::ZeroOrMore, cl::desc("Print information about actions taken"),
...
+  default:
+    llvm_unreachable("Only expecting errors");
+  }
+
+  DiagnosticPrinterRawOStream DP(errs());
+  DI.print(DP);
+  errs() << '\n';
+}
+
+static bool emitLinkerError(LLVMContext &Context, const Twine &Message) {
```
Comment (emankov) [inline src/driver/AmdCompiler.cpp]: I'd prefer avoiding static, otherwise LGTM.

**Pull request merged**